include/picongpu/particles/atomicPhysics/ionizationPotentialDepression/StewartPyattIPD.hpp

```diff
@@ -183,62 +198,79 @@ namespace picongpu::particles::atomicPhysics::ionizationPotentialDepression
         };
 
-        /** calculate ionization potential depression
+        /** get the super cell constant input
          *
+         * @param superCellFieldIdx index of superCell in superCellField(without guards)
+         * @param debyeLengthBox deviceDataBox giving access to the local debye length for all local superCells,
+         *  sim.unit.length(), not weighted
          * @param temperatureEnergyBox deviceDataBox giving access to the local temperature * k_Boltzman for all
          *  local superCells, in sim.unit.mass() * sim.unit.length()^2 / sim.unit.time()^2, not weighted
          * @param zStarBox deviceDataBox giving access to the local z^Star value, = average(q^2) / average(q),
          *  for all local superCells, unitless, not weighted
-         * @param debyeLengthBox deviceDataBox giving access to the local debye length for all local superCells,
-         *  sim.unit.length(), not weighted
-         * @param superCellFieldIdx index of superCell in superCellField(without guards)
-         *
-         * @return unit: eV, not weighted
          */
-        template<
-            uint8_t T_atomicNumber,
-            typename T_DebyeLengthBox,
-            typename T_TemperatureEnergyBox,
-            typename T_ZStarBox>
-        HDINLINE static float_X calculateIPD(
+        template<typename T_DebyeLengthBox, typename T_TemperatureEnergyBox, typename T_ZStarBox>
+        HDINLINE static SuperCellConstantInput getSuperCellConstantInput(
             pmacc::DataSpace<simDim> const superCellFieldIdx,
             T_DebyeLengthBox const debyeLengthBox,
             T_TemperatureEnergyBox const temperatureEnergyBox,
             T_ZStarBox const zStarBox)
         {
-            // eV/(sim.unit.energy())
-            constexpr float_X eV = sim.pic.get_eV();
+            // eV, not weighted
+            float_X temperatureTimesk_Boltzman = temperatureEnergyBox(superCellFieldIdx);
+            // UNIT_LENGTH, not weighted
+            float_X debyeLength = debyeLengthBox(superCellFieldIdx);
+            // unitless, not weighted
+            float_X zStar = zStarBox(superCellFieldIdx);
+
+            auto superCellConstantInput = SuperCellConstantInput();
+            superCellConstantInput.temperatureTimesk_Boltzman = temperatureTimesk_Boltzman;
+            superCellConstantInput.debyeLength = debyeLength;
+            superCellConstantInput.zStar = zStar;
 
-            // eV/(sim.unit.mass() * sim.unit.length()^2 / sim.unit.time()^2) * unitless * sim.unit.charge()^2
-            //  / ( unitless * sim.unit.charge()^2 * sim.unit.time()^2 / (sim.unit.length()^3 * sim.unit.mass()))
-            // = eV * sim.unit.time()^2 * sim.unit.mass()^(-1) * sim.unit.length()^(-2) * sim.unit.charge()^2 *
-            // sim.unit.charge()^(-2)
-            //  * sim.unit.time()^(-2) * sim.unit.length()^3 * sim.unit.mass()^1 = eV * sim.unit.length()
-            // eV * sim.unit.length()
+            return superCellConstantInput;
```

Comment:
```suggestion
            return {temperatureTimesk_Boltzman, debyeLength, zStar};
```
Can simply be this, but if you are worried that someone will change the order of the struct members and silently run into undiagnosable problems, you can add a constructor as suggested above

Reply:
used the constructor version